WinImageBuilder.psm1

```diff
@@ -713,60 +724,81 @@ function Compress-Image {
         [Parameter(Mandatory=$true)]
         [string]$VirtualDiskPath,
         [Parameter(Mandatory=$true)]
-        [string]$ImagePath
+        [string]$ImagePath,
+        [Parameter(Mandatory=$true)]
+        [string[]]$compressionFormats,
+        [parameter(Mandatory=$false)]
+        [string]$ZipPassword
     )
     if (!(Test-Path $VirtualDiskPath)) {
         Throw "$VirtualDiskPath not found"
     }
-    $tmpName = '{0}.tar' -f @((Get-PathWithoutExtension($ImagePath)))
-
     $7zip = Get-7zipPath
     $pigz = Get-PigzPath
+    $tmpName = (Get-Item $VirtualDiskPath).Name
+    $compressionFormats = $compressionFormats.split(".")
     try {
-        Write-Host "Archiving $VirtualDiskPath to tarfile $tmpName"
-        Push-Location ([System.IO.Path]::GetDirectoryName((Resolve-path $VirtualDiskPath).Path))
-        try {
-            # Avoid storing the full path in the archive
-            $imageFileName = (Get-Item $VirtualDiskPath).Name
-            Write-Host "Creating tar archive..."
-            & $7zip a -ttar $tmpName $imageFileName
-            if ($LASTEXITCODE) {
-                if ((Test-Path $imageFileName)) {
-                    Remove-Item -Force $imageFileName
-                }
-                throw "7za.exe failed while creating tar file for image: $tmpName"
+        foreach ($compresionFormat in $compressionFormats) {
+            if ($compresionFormat -eq "tar") {
+                $tmpName = '{0}.tar' -f @((Get-PathWithoutExtension($ImagePath)))
+                Write-Host "Archiving $VirtualDiskPath to tarfile $tmpName"
+                    # Avoid storing the full path in the archive
+                    $imageFileName = (Get-Item $VirtualDiskPath).Name
+                    Write-Host "Creating tar archive..."
+                    & $7zip a -ttar $tmpName $imageFileName

```

Comment:
Is mandatory to be in the directory you need
Push-Location ([System.IO.Path]::GetDirectoryName((Resolve-path $VirtualDiskPath).Path))

Reply:
add a popd too